src/solvation/hnd_coschg.F

```diff
@@ -1220,14 +1225,16 @@ subroutine hnd_coschg(g_dens,ndens,rtdb,geom,basis,nat,nefc,
       call ga_sync()
 c
       if (do_cosmo_vem.ne.0.and.istep_cosmo_vem.eq.0) then
-       if(.not.rtdb_get(rtdb,'cosmo:g_vem',mt_int,3,g_vem)) then
-        status = nga_create(mt_dbl, 1, nefc, 'V_GS', nefc, g_vem(1))
-        status = nga_create(mt_dbl, 1, nefc, 'Q_GS', nefc, g_vem(2))
-        status = nga_create(mt_dbl, 1, nefc, 'Q_neq', nefc, g_vem(3))
-        if(.not.rtdb_put(rtdb,'cosmo:g_vem',mt_int,3,g_vem))
-     $  call errquit('hnd_coschg: cannot put g_vem in rtdb',
-     $  0,rtdb_err)
-       endif
+         if(.not.rtdb_get(rtdb,'cosmo:g_vem',mt_int,3,g_vem)) 
+     c        call hnd_create_gvems(rtdb,nefc,g_vem)
+         call ga_check_handle(g_vem(1), 'gvem1')
+         call nga_inquire(g_vem(1), type1, ndim, dim1)
+      if(dim1.ne.nefc) then
+         if(ga_nodeid().eq.0) write(luout,*)
+     W        ' W: dims1.ne.nefc. creating g_vems'
+         call hnd_create_gvems(rtdb,nefc,g_vem)
```

Comment:
We need to destroy the previous `g_vem` before calling `hnd_create_gvems`, right?

Reply:
> We need to destroy the previous `g_vem` before calling `hnd_create_gvems`, right?

You are absolutely right.
Thanks for catching this.